src/api/app/models/token/service.rb

```diff
@@ -4,6 +4,8 @@ def self.token_name
   end
 
   def call(_options)
+    # we can not work on remote sources
+    raise ActiveRecord::RecordNotFound if @token.package_from_association_or_params.nil?
```

Comment:
You also will never get here because the policy for this Token will not let you. See [here](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/policies/token/service_policy.rb#L10).

Reply:
this is only for the source service where it is okay.

release and rebuild have different policies.

(it would been better not to ignore the package_find_options now to be able to have a single policy and to have execution, lookup and permission in a single file. seperating the package lookup from permission handling adds security risks IMHO since it needs to be kept in sync).